compiler/symcc.in

```diff
@@ -38,10 +38,11 @@ if [ $# -eq 0 ]; then
     exit 1
 fi
 
-exec $compiler                                  \
+exec "$compiler"                                \
      @CLANG_LOAD_PASS@"$pass"                   \
      "$@"                                       \
      -L"$runtime_dir"                           \
-     -lSymRuntime                               \
+     -lsymcc-rt                                 \
      -Wl,-rpath,"$runtime_dir"                  \
+     -lstdc++                                   \
```

Comment:
Why do we link against `libstdc++` here? I guess this is to support the static version of the runtime; if that's the case then I'm not convinced that it's the right approach. We can't be sure that the runtime has been built with `libstdc++` (vs some other C++ runtime like `libc++`) :thinking:

Reply:
Indeed, I don't really like this fix as well.
Without this flag, I get this error:
```
Symbolizer module instrumentation
Symbolizing function main
/usr/bin/ld: /tmp/sample-f61b56.o: undefined reference to symbol '_ZSt3cin@@GLIBCXX_3.4'
/usr/bin/ld: /usr/lib/libstdc++.so.6: error adding symbols: DSO missing from command line
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
There is a [StackOverflow post](https://stackoverflow.com/questions/33263288/libstdc-dso-missing-from-command-line) about this issue, maybe we could use g++ instead of gcc as suggested?